augur/tasks/github/util/github_paginator.py

```diff
@@ -7,16 +7,52 @@
 import asyncio
 import datetime
 import logging
-
+from sqlalchemy.sql import text
 
 from typing import List, Optional, Union, Generator, Tuple
 from urllib.parse import urlencode, urlparse, parse_qs, urlunparse
 from enum import Enum
 
 
 from augur.tasks.github.util.github_random_key_auth import GithubRandomKeyAuth
+from augur.application.db.session import DatabaseSession
 from augur.tasks.github.util.util import parse_json_response
 
+def update_moved_repo(old_url, new_url, logger):
+
+    with DatabaseSession(logger) as session:
+
+        # check if new url already exists if so delete old url and return
+        new_url_query = text(f"""SELECT repo_id FROM repo WHERE repo_git={new_url}""")
+        result = session.fetchall_data_from_sql_text(new_url_query)
+
+        if len(result) > 0:
+
+            delete_repo_query = text(f"""DELETE FROM augur_data.commits where repo_id={repo_id};
+                                        DELETE FROM augur_data.issue_assignees where repo_id ={repo_id};
+                                        DELETE FROM augur_data.issue_events where repo_id={repo_id};
+                                        DELETE FROM augur_data.issue_labels where repo_id={repo_id};
+                                        DELETE FROM augur_data.issue_message_ref where repo_id={repo_id};
+                                        DELETE FROM augur_data.pull_request_assignees where repo_id={repo_id};
+                                        DELETE FROM augur_data.pull_request_commits where repo_id={repo_id};
+                                        DELETE FROM augur_data.pull_request_events where repo_id={repo_id};
+                                        DELETE FROM augur_data.pull_request_files where repo_id={repo_id};
+                                        DELETE FROM augur_data.pull_request_meta where repo_id={repo_id};
+                                        DELETE FROM augur_data.releases where repo_id={repo_id};
+                                        DELETE FROM augur_data.pull_request_message_ref where repo_id={repo_id};
+                                        DELETE FROM augur_operations.user_repos where repo_id={repo_id};
+                                        DELETE FROM augur_data.pull_request_labels where repo_id={repo_id};
+                                        DELETE FROM repo where repo_id={repo_id};
+                                """)
+            session.execute_sql(delete_repo_query)
+            return 
+
+        # update the old url to the new url 
+        query = text("""SELECT * FROM augur_data.repo WHERE repo_git='old_url' FOR UPDATE;
```

Comment:
@ABrain7710 : I think we also need to set repo_status='New', repo_path=NULL, and repo_name=NULL here.

Reply:
Presuming we are deleting everything, which is what it looks like .. then we'd be needing to also reset the repo in this way.